pkg/auth/dependency/interaction/adaptors/identity.go

```diff
@@ -32,7 +32,13 @@ type OAuthIdentityProvider interface {
 		profile map[string]interface{},
 		claims map[string]interface{},
 	) *oauth.Identity
+	NewForUpdate(
+		id string,
+		userID string,
+		profile map[string]interface{},
+	) (*oauth.Identity, error)
```

Comment:
This method seems too specific. The caller can just use `Get` and assign to `UserProfile`.